modules/sw360/sw360-client/src/main/java/org/eclipse/sw360/antenna/sw360/rest/resource/releases/SW360Release.java

```diff
@@ -338,12 +340,26 @@ public SW360Release mergeWith(SW360Release releaseWithPrecedence) {
         if (componentIdWithPrecedence != null && !componentIdWithPrecedence.getHref().isEmpty()) {
             get_Links().setSelfComponent(componentIdWithPrecedence);
         }
+        final List<SW360SparseAttachment> releaseWithPrecedenceAttachments = releaseWithPrecedence.get_Embedded().getAttachments();
+        if (releaseWithPrecedenceAttachments != null && !releaseWithPrecedenceAttachments.isEmpty()) {
+            if (get_Embedded().getAttachments() == null || get_Embedded().getAttachments().isEmpty()) {
+                get_Embedded().setAttachments(releaseWithPrecedenceAttachments);
+            } else {
+                get_Embedded().setAttachments(mergeAttachments(get_Embedded().getAttachments(), releaseWithPrecedenceAttachments));
+            }
+        }
         externalIds.putAll(releaseWithPrecedence.externalIds);
         additionalData.putAll(releaseWithPrecedence.additionalData);
 
         return this;
     }
 
+    private List<SW360SparseAttachment> mergeAttachments(List<SW360SparseAttachment> attachments, List<SW360SparseAttachment> releaseWithPrecedenceAttachments) {
+        return Stream.concat(attachments.stream(), releaseWithPrecedenceAttachments.stream())
+                .distinct()
+                .collect(Collectors.toList());
+    }
+
```

Comment:
One proposal: As duplicate attachments are obviously not allowed, it could be an option to store them in SW360ReleaseEmbedded in a Set rather than a List. Then a merge is straight-forward: just construct a union set from both releases involved.